components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/util/OAuth2UtilTest.java

```diff
@@ -211,27 +216,31 @@ public void testClearClientTenantId() throws Exception {
     @DataProvider(name = "BuildScopeString")
     public Object[][] buildScopeString() {
```

Comment:
Can we also include a test case,
new String[] { "Z" , "Y", "X"} which gives the output "X Y Z" (ie. Scope strings are sorted before they are combined)

Reply:
Updated the PR with an extra test case.